src/Integrations/Integrations/Frankenphp/FrankenphpIntegration.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+namespace DDTrace\Integrations\Frankenphp;
+
+use DDTrace\HookData;
+use DDTrace\Integrations\Integration;
+use DDTrace\SpanStack;
+use DDTrace\Tag;
+use DDTrace\Type;
+
+use function DDTrace\consume_distributed_tracing_headers;
+
+class FrankenphpIntegration extends Integration
+{
+    const NAME = 'frankenphp';
+
+    public function getName()
+    {
+        return self::NAME;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function requiresExplicitTraceAnalyticsEnabling()
+    {
+        return false;
+    }
+
+    public function init()
+    {
+        $integration = $this;
+
+        ini_set("datadog.trace.auto_flush_enabled", 1);
+        ini_set("datadog.trace.generate_root_span", 0);
+
+        \DDTrace\install_hook('frankenphp_handle_request', function (HookData $hook) use ($integration) {
+            $hook->data = \DDTrace\install_hook(
+                $hook->args[0],
+                function (HookData $hook) use ($integration) {
+                    $rootSpan = $hook->span(new SpanStack());
+                    $rootSpan->name = "web.request";
+                    $rootSpan->service = \ddtrace_config_app_name('frankenphp');
+                    $rootSpan->type = Type::WEB_SERVLET;
+                    $rootSpan->meta[Tag::COMPONENT] = FrankenphpIntegration::NAME;
+                    $rootSpan->meta[Tag::SPAN_KIND] = Tag::SPAN_KIND_VALUE_SERVER;
+                    unset($rootSpan->meta["closure.declaration"]);
+                    $integration->addTraceAnalyticsIfEnabled($rootSpan);
+
+                    consume_distributed_tracing_headers(null);
+                }, function (HookData $hook) {
+                    var_dump($hook->span());
+                }
+            );
+        }, function (HookData $hook) {
+            \DDTrace\remove_hook($hook->data);
+        });
```

Comment:
_I'm wondering out loud_

I understand that we want the hook on the callback to be set only once, the `frankenphp_handle_request` being called on each iteration of the main loop.

However, I ponder whether it is better to:
- Install and remove the hook on the callback after the request is dispatched, hence having this overhead for basically all requests (although this happens _outside_ a request)
  - _This is what's currently done_

OR

- Don't remove the callback hook; instead, only install it if it isn't already set. I guess we could check this through a WeakMap on the Closure. This way, I think we could also use `hook_function` on `frankenphp_handle_request`, which I believe to be less performance hungry...

However, it is also possible that this change would be so negligible that it isn't meaningful...

Have you considered this way?

Reply:
Changed it. I don't think it makes a big difference, but anyway, why not :-)